src/ast.rs

```diff
@@ -173,6 +194,24 @@ impl<F, TraceArgs> Circuit<F, TraceArgs> {
 
         Rc::clone(step_rc)
     }
+
+    pub fn expose_first(&mut self, signal: Signal) {
```

Comment:
In the AST you just need to store the information, then the compiler will convert it to column, rotation

Reply:
So you can store with the  ExposeOffset enum. And perhaps you don't need three methods then.